src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -366,16 +376,53 @@ private void ValidateEnumDependencies(JsonSchema schema, JObject map, List<Error
                             if (allowLists.TryGetValue(dependentFieldValue, out var allowList) &&
                                 Array.IndexOf(allowList, fieldValue) == -1)
                             {
-                                errors.Add(Errors.InvalidPairedAttribute(JsonUtility.GetSourceInfo(map), fieldName, fieldValue, dependentFieldName, dependentFieldValue));
+                                errors.Add(GenerateAdditionalError(schema, dependentFieldName, Errors.InvalidPairedAttribute(JsonUtility.GetSourceInfo(map), fieldName, fieldValue, dependentFieldName, dependentFieldValue), map));
                             }
                         }
                         else
                         {
-                            errors.Add(Errors.MissingPairedAttribute(JsonUtility.GetSourceInfo(map), fieldName, dependentFieldName));
+                            errors.Add(GenerateAdditionalError(schema, fieldName, Errors.MissingPairedAttribute(JsonUtility.GetSourceInfo(map), fieldName, dependentFieldName), map));
+                        }
+                    }
+                }
+            }
+        }
+
+        private Error GenerateAdditionalError(JsonSchema schema, string name, Error baseError, JToken token)
+        {
+            if (!string.IsNullOrEmpty(name) && schema.AdditionalErrors.TryGetValue(name, out var attributeAdditionalErrors) && attributeAdditionalErrors.TryGetValue(baseError.Code, out var additionalError))
+            {
+                return new Error(
+                    additionalError.Severity == ErrorLevel.Off ? baseError.Level : additionalError.Severity,
+                    string.IsNullOrEmpty(additionalError.Code) ? baseError.Code : additionalError.Code,
+                    string.IsNullOrEmpty(additionalError.Message) ? baseError.Message : additionalError.Message,
```

Comment:
so it is re-placing the message instead of appending more message? If so, `additionalErrors` might not be an appropriate name, it makes me feel like it'll generate more errors and more messages.

Reply:
agree, it look like "OverwriteErrors"

Reply:
could't agree any more